axelor-human-resource/src/main/java/com/axelor/apps/hr/service/timesheet/TimesheetServiceImpl.java

```diff
@@ -1052,4 +1059,47 @@ public void generateLinesFromProjectPlanning(Timesheet timesheet, Boolean realHo
     }
     return planningList;
   }
+
+  @Override
+  public void prefillLines(Timesheet timesheet) throws AxelorException {
+    PublicHolidayService holidayService = Beans.get(PublicHolidayService.class);
+    LeaveService leaveService = Beans.get(LeaveService.class);
+    WeeklyPlanningService weeklyPlanningService = Beans.get(WeeklyPlanningService.class);
+    AppTimesheet appTimesheet = appHumanResourceService.getAppTimesheet();
+
+    LocalDate fromDate = timesheet.getFromDate();
+    LocalDate toDate = timesheet.getToDate();
+    User user = timesheet.getUser();
+    WeeklyPlanning weeklyPlanning = user.getEmployee().getWeeklyPlanning();
+    EventsPlanning holidayPlanning = user.getEmployee().getPublicHolidayEventsPlanning();
+
+    for (LocalDate date = fromDate; !date.isAfter(toDate); date = date.plusDays(1)) {
+      BigDecimal dayValueInHours =
+          weeklyPlanningService.getWorkingDayValueInHours(
+              weeklyPlanning, date, LocalTime.MIN, LocalTime.MAX);
+
+      if (appTimesheet.getCreateLinesForHolidays()
+          && holidayService.checkPublicHolidayDay(date, holidayPlanning)) {
+        prefillLine(
+            timesheet, user, date, dayValueInHours, I18n.get(IExceptionMessage.TIMESHEET_HOLIDAY));
+
+      } else if (appTimesheet.getCreateLinesForLeaves()) {
+        LeaveRequest leave = leaveService.getLeave(user, date);
+        if (leave != null) {
+          BigDecimal hours = leaveService.computeDuration(leave);
+          if (leave.getLeaveLine().getLeaveReason().getUnitSelect()
+              == LeaveReasonRepository.UNIT_SELECT_DAYS) {
+            hours = hours.multiply(dayValueInHours);
+          }
+          prefillLine(
+              timesheet, user, date, hours, I18n.get(IExceptionMessage.TIMESHEET_DAY_LEAVE));
+        }
+      }
+    }
+  }
+
+  protected void prefillLine(
```

Comment:
Maybe it's better to overload createTimesheetLinePeriod with these 5 parameters.

Reply:
Done.